File: benchmarks/oopsla25/subjects/061.c
#include <assert.h>
#include <fcntl.h>
#include <stdio.h>
#include <stdlib.h>
#include <string.h>
#include <unistd.h>

char* f061(const char *s) {
    assert(strlen(s) <= 3);
    
    static char result[4];
    result[0] = s[0];
    result[1] = s[1];
    result[2] = s[2];
    result[3] = '\0';

    return result;
}

int main(int argc, char *argv[]) {
  char my_string[10240] = {0};
  if (argc == 1) {
    read(fileno(stdin), my_string, 10240);
  } else {
    int fd = open(argv[1], O_RDONLY);
    read(fd, my_string, 10240);
    close(fd);
  }
  char *s = f061(my_string);
  printf("%s\n", s);
  return 0;
}

File: benchmarks/oopsla25/subjects/061.py
import sys

def f061(s: str) -> str:
  assert len(s) <= 3
  c0 = s[0]
  c1 = s[1]
  c2 = s[2]
  return c0 + c1 + c2

if __name__ == '__main__':
  my_string = ""
  if len(sys.argv) == 1:
    my_string = sys.stdin.read()
  else:
    fd = open(sys.argv[1], 'r')
    my_string = fd.read()
    fd.close()
  
  print(f061(my_string))
